### ISSUE TITLE
DataTable with expansion does not work with sticky header

### ISSUE THREAD

**Issue by gmoehler** [carbon-design-system/carbon#3912]
## Brief description
DataTables with expansion are not yet supported very well with sticky header. The pulldown is somewhat mis-aligned with the rest of the row.

## What package(s) are you using?
latest

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.
When using the appropriate styles in Vanilla JS or the new stickyHeader flag in the react component (https://github.com/carbon-design-system/carbon/issues/3875) along with expansion the row does not look properly (see screenshot)

> Is this issue related to a specific component?
DataTable

> What did you expect to happen? What happened instead? What would you like to
> see changed?
row should look the same with or without sticky header

> What browser are you working in?
chrome / firefox

> What version of the Carbon Design System are you using?
latest 

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?
IBM Cloud SQL Query

## Steps to reproduce the issue

1. Select *DataTable with expansion* in storyboard
2. In Knows turn on the sticky header option
3. The table rows are no longer showing up properly 

## Additional information

- Screenshots 
<img width="640" alt="Screenshot 2019-09-05 at 15 02 23" src="https://user-images.githubusercontent.com/17567047/64344297-7a263080-cfee-11e9-9c8e-78d4a4b24985.png">

**Comment by tw15egan** [comment]
Seems to still be an issue

**Comment by sstrubberg** [comment]
Is this relevant @gmoehler ?

**Comment by gmoehler** [comment]
@sstrubberg sorry for the late response. in fact we have been using a sticky table unchanged since that time in the product, and had to add a number of css styles to make it working. Unfortunatelly, the current [carbon 10 react storybook](https://v7-react.carbondesignsystem.com/?path=/story/components-datatable-expansion--usage) no longer has the option to set parameters via `knobs`, so it is hard to easily prove its current state. So I _assume_ it is still relevant.

**Comment by tyranasaurus** [comment]
This is still an issue.

**Comment by tay1orjones** [comment]
Yeah, [here's an example](https://react.carbondesignsystem.com/?path=/story/components-datatable-expansion--playground&args=stickyHeader:!true) of it in the current storybook



![Image](https://github.com/carbon-design-system/carbon/assets/3360588/cbfedcaf-4f32-451e-a03d-0886845bd6c0)

**Comment by tay1orjones** [comment]
@gmoehler could you share some of the overrides you've made to get this working? Would love to get those fixes upstream here if possible!
